src/components/Compare/CompareTable.tsx

```diff
@@ -0,0 +1,191 @@
+import React, { useState } from 'react';
+import { TableBody } from '@material-ui/core';
+import {
+  Wrapper,
+  StyledTable,
+  ArrowContainer,
+  TableHeadContainer,
+  Row,
+  Cell,
+  Footer,
+  ViewAllLink,
+  Header,
+} from 'components/Compare/Compare.style';
+import CompareTableRow from 'components/Compare/CompareTableRow';
+import HeaderCell from 'components/Compare/HeaderCell';
+import { getLocationNames, Location } from 'common/locations';
+import {
+  stateSummary,
+  countySummary,
+  LocationSummary,
+} from 'common/location_summaries';
+import { Metric } from 'common/metric';
+import ExpandLessIcon from '@material-ui/icons/ExpandLess';
+import ExpandMoreIcon from '@material-ui/icons/ExpandMore';
+import { COLOR_MAP } from 'common/colors';
+import { fail } from 'common/utils';
+
+const locations: any = getLocationNames();
+
+export interface SummaryForCompare {
+  locationInfo: Location;
+  metricsInfo: LocationSummary;
+}
+
+const CompareTable = (props: {
+  stateId?: string;
+  stateName?: string;
+  county?: any | null;
+  setShowModal: any;
+  isModal?: Boolean;
+  locationsViewable?: number;
+  isLocationPage?: Boolean;
+  isHomepage?: Boolean;
+}) => {
```

Comment:
I think it might be better to refactor this component to receive an array of locations with metrics and overall stats, that way we don't need to pass state, county, `isLocationPage`, etc. The parent component knows what is the current location (home page, state, or county page) so it has the context of which locations should be shown.

In this implementation, the parent passes some information `isLocationPage` and location information (`stateId?`, `stateName?`, `county?` and then the `CompareTable` component needs to calculate the items to show.

```tsx
<CompareTable
  locations={locationList}
  numLocations={5}
  title={`${locationName} State`}
/>
```

Reply:
Moved the location arrays out of CompareTable and into the top levels (Homepage + ChartsHolder), and removed stateId and isLocationPage props. Adding a TODO to refactor further and remove the need for the other page-specific props.